More updates for Data Prepper 2.0 (#1469)
opensearch-project/documentation-website: _clients/data-prepper/data-prepper-reference.md

privateKeyPassword | No | String | Password for private key within keystore. Optional, defaults to empty string.
serverPort | No | Integer | Port number to use for server APIs. Defaults to 4900
metricRegistries | No | List | Metrics registries for publishing the generated metrics. Currently supports Prometheus and CloudWatch. Defaults to Prometheus.

> metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is 3. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, the value will overwrite those set through the above methods.

Reviewer: > Its value could also be set through the

I recommend using active voice.

> Alternatively, administrators can set this value through the

Author: Thanks! These has been addressed in [ca1e7da](https://github.com/opensearch-project/documentation-website/pull/1469/commits/ca1e7da0795d2621ec2dd11045155b545cc8fa99).

> metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is 3. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, the value will overwrite those set through the above methods.

Reviewer 2: ```suggestion
metricTags | No | Map | Key-value pairs as common metric tags to metric registries. The maximum number of pairs is three. Note that `serviceName` is a reserved tag key with `DataPrepper` as default tag value. Its value could also be set through the environment variable `DATAPREPPER_SERVICE_NAME`. If `serviceName` is defined in `metricTags`, that value overwrites those set through the above methods.
```

Author: Thanks! All addressed in [ca1e7da](https://github.com/opensearch-project/documentation-website/pull/1469/commits/ca1e7da0795d2621ec2dd11045155b545cc8fa99).

> authentication | No | Object | Authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server will not perform authentication.

Reviewer 2: ```suggestion
authentication | No | Object | Authentication configuration. Valid option is `http_basic` with `username` and `password` properties. If not defined, the server does not perform authentication.
```

processorShutdownTimeout | No | Duration | Time given to processors to clear any in-flight data and gracefully shutdown. Default is 30s.
sinkShutdownTimeout | No | Duration | Time given to sinks to clear any in-flight data and gracefully shutdown. Default is 30s.

> peer_forwarder | No | Ojbect | Peer forwarder configurations. See [Peer forwarder options](#peer-forwarder-options) for more details.

Reviewer: Spelling: "Ojbect" -> "Object"

port | No | Integer | The port OTel trace source is running on. Default is `21890`.
request_timeout | No | Integer | The request timeout in milliseconds. Default is `10_000`.
health_check_service | No | Boolean | Enables a gRPC health check service under `grpc.health.v1/Health/Check`. Default is `false`.

> unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. This option is ignored if no authentication is defined. Default is `false`.

Reviewer: > Whether authentication is required

Perhaps add a verb here:

> Determines whether authentication is required

> unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. This option is ignored if no authentication is defined. Default is `false`.

Reviewer 2: ```suggestion
unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```

Option | Required | Type | Description
:--- | :--- | :--- | :---
port | No | Integer | The port the source is running on. Default is `2021`. Valid options are between `0` and `65535`.
health_check_service | No | Boolean | Enables health check service on `/health` endpoint on the defined port. Default is `false`.

> unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. This option is ignored if no authentication is defined. Default is `false`.

Reviewer: Similar comment as above.

> unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. This option is ignored if no authentication is defined. Default is `false`.

Reviewer 2: ```suggestion
unauthenticated_health_check | No | Boolean | Whether authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```

> unauthenticated_health_check | No | Boolean | Determines whether authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.

Reviewer 3: ```suggestion
unauthenticated_health_check | No | Boolean | Determines whether or not authentication is required on the health check endpoint. Data Prepper ignores this option if no authentication is defined. Default is `false`.
```

connect_timeout | No | Integer | The timeout in milliseconds used when requesting a connection from the connection manager. A timeout value of zero is interpreted as an infinite timeout. If this timeout value is either negative or not set, the underlying Apache HttpClient would rely on operating system settings for managing connection timeouts.
insecure | No | Boolean | Whether to verify SSL certificates. If set to true, CA certificate verification is disabled and insecure HTTP requests are sent instead. Default is `false`.
proxy | No | String | The address of a [forward HTTP proxy server](https://en.wikipedia.org/wiki/Proxy_server). The format is "&lt;host name or IP&gt;:&lt;port&gt;". Examples: "example.com:8100", "http://example.com:8100", "112.112.112.112:8100". Port number cannot be omitted.

> index | Conditionally | String | Name of the index to export to. Applicable and required only when the `index_type` is `custom`.

Reviewer 3: ```suggestion
index | Conditionally | String | Specifies the export index name. Applicable and required only when the `index_type` is `custom`.
```

> index | Conditionally | String | Name of the index to export to. Applicable and required only when the `index_type` is `custom`.

Reviewer 3: ```suggestion
index | Conditionally | String | Name of the export index. Applicable and required only when the `index_type` is `custom`.
```

> index | Conditionally | String | Name of the index to export to. Applicable and required only if `index_type` is explicitly `custom` or defaults to `custom`.

Reviewer: > Applicable and required only if `index_type` is explicitly `custom` or defaults to `custom`.

Perhaps reword to:

> Applicable and required only when the `index_type` is `custom`.
